index/secondary_index.cc

```diff
@@ -157,7 +206,23 @@ sstring target_parser::serialize_targets(const std::vector<::shared_ptr<cql3::st
     };
 
     if (targets.size() == 1 && std::holds_alternative<index_target::single_column>(targets.front()->value)) {
-        return std::get<index_target::single_column>(targets.front()->value)->to_string();
+        auto& target = targets.front();
+        auto single_target = std::get<index_target::single_column>(target->value);
+        switch (target->type) {
+            case index_target::target_type::regular_values:
```

Comment:
nitpick: add `[[fallthrough]];` after each fallthrough case. That applies to this whole series, so I'll just mention it here once.